sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/mongodb.yaml

```diff
@@ -0,0 +1,118 @@
+          # Certain special characters need to be percent-encoded so we can break matching when we find them.
+          pattern: "\\bmongo(?:db)?(?:\\+\\w+)?://[^:@/ ]:([^@/ ]+)@"
+        post:
+          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
+          patternNot: "(?i)^(\\$[a-z_]*)?(db|my)?_?pass(word|wd)?|^\\[[\\w\\t \\-]+\\]$|^<[\\w\\t \\-]+>$|^%s$|^\\$\\w+$|^\\{+[^}]*\\}+$|^\\$\\(\\w+\\)$"
+      examples:
+        - text: >
+            backend "pg" {
+              conn_str = "mongo://user:pass@database.example/terraform_backend"
+            }
+          containsSecret: false
+        - text: >
+            connStr = fmt.Sprintf("mongodb://%s:%s@:%s/%s%ssslmode=%s&host=%s",
+              url.PathEscape(dbUser), url.PathEscape(dbPasswd), port, dbName, dbParam, dbsslMode, host)
+          containsSecret: false
+        - text: >
+            if (!newConfig.DATABASE_URL) {
+              const encodedUser = encodeURIComponent(newConfig.POSTHOG_DB_USER)
+              const encodedPassword = encodeURIComponent(newConfig.POSTHOG_DB_PASSWORD)
+              newConfig.DATABASE_URL = `mongodb://${encodedUser}:${encodedPassword}@${newConfig.POSTHOG_MONGO_HOST}:${newConfig.POSTHOG_MONGO_PORT}/${newConfig.POSTHOG_DB_NAME}`
+            }
+          containsSecret: false
+        - text: >
+            SqlSettingsDefaultDataSource = "mongodb+srv://mmuser:mostest@localhost/mattermost_test?sslmode=disable&connect_timeout=10&binary_parameters=yes"
+          containsSecret: true
+          match: mostest
+        - text: >
+            passwordFile := fs.String("password", "../../../tools/secrets/password.txt", "password file")
+            databasePrefix := fs.String("database-prefix", "mongodb://mongo:mongo_password_padded_for_security@localhost:27017/ocr2vrf-test", "database prefix")
+            databaseSuffixes := fs.String("database-suffixes", "sslmode=disable", "database parameters to be added")
+          containsSecret: true
+          match: mongo_password_padded_for_security
+        - text: >
+            driver: mongodb
+            dsn: mongodb://foouser:foopass@localhost:27017/testdb?sslmode=disable
+            table: footable
+          containsSecret: true
+          match: foopass
+        - text: >
+            AIRFLOW_CONN_METADATA_DB=mongodb+srv://airflow:airflow@mongo:27017/airflow
+            AIRFLOW_VAR__METADATA_DB_SCHEMA=airflow
+          containsSecret: true
+          match: airflow
+
+    - id: mongodb-cli-unquoted
+      metadata:
+        name: MongoDB database passwords should not be disclosed
+      detection:
+        pre:
+          include:
+            content:
+              - mongodb
+          reject:
+            ext:
+              - .md
+              - .html
+              - .example
+              - .template
+        matching:
+          pattern: "\\bmongodb\\b.{1,50}--password\\s+(?![\"'])([^\\s]+)"
```

Comment:
You're right about `mongodb`. I found a web page with bad info and got lots of matches because of `mongodb://` URLs.

It turns out that there are several tools that start with `mongo` that use the same syntax. The updated regex will support all of them.

I've also discovered that `-p` is an alias for `--password` and will need a small negative lookbehind to make that work correctly. The negative lookahead near the capture group will be removed.